File: ios/CustomMethods.m
//
//  CustomMethods.m
//  native_module
//
//  Created by mac on 07/06/24.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(CustomMethods,NSObject)
RCT_EXTERN_METHOD(MyMethod : (NSString*) param)

@end

File: ios/CustomMethods.swift
import Foundation
import UIKit

@objc(CustomMethods)
class CustomMethods: NSObject {
    @objc static func requiresMainQueueSetup() -> Bool {
        return true
    }

    @objc public func MyMethod(_ param: String) {
        DispatchQueue.main.async {
            if let rootViewController = UIApplication.shared.keyWindow?.rootViewController {
                let buttonViewController = ButtonViewController()
                rootViewController.present(buttonViewController, animated: true, completion: nil)
            }
        }
    }
}
